File: apps/packages/core/core_models/lib/src/syllabus_item.dart
class SyllabusItem {
  const SyllabusItem({
    required this.id,
    required this.code,
    required this.title,
    this.parentId,
    this.children = const [],
  });

  final String id;
  final String? parentId;
  final String code;
  final String title;
  final List<SyllabusItem> children;
}

File: backend/src/main/java/com/preppy/taxonomy/SyllabusItem.java
package com.preppy.taxonomy;

import java.util.List;
import java.util.UUID;

/**
 * Single node in the syllabus tree. Self-referential through {@link #children}.
 */
public record SyllabusItem(UUID id, UUID parentId, String code, String title, List<SyllabusItem> children) {
}
